example-app/src/main/java/com/adyen/checkout/example/ui/googlepay/compose/SessionsGooglePayActivity.kt

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2023 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by josephj on 13/12/2023.
+ */
+
+package com.adyen.checkout.example.ui.googlepay.compose
+
+import android.content.Intent
+import android.os.Bundle
+import androidx.activity.compose.setContent
+import androidx.activity.viewModels
+import androidx.appcompat.app.AppCompatActivity
+import androidx.compose.foundation.isSystemInDarkTheme
+import androidx.core.view.WindowCompat
+import com.adyen.checkout.example.ui.googlepay.GooglePayFragment
+import com.adyen.checkout.example.ui.theme.ExampleTheme
+import com.adyen.checkout.example.ui.theme.NightTheme
+import com.adyen.checkout.example.ui.theme.NightThemeRepository
+import com.adyen.checkout.redirect.RedirectComponent
+import dagger.hilt.android.AndroidEntryPoint
+import javax.inject.Inject
+
+@AndroidEntryPoint
+class SessionsGooglePayActivity : AppCompatActivity() {
+
+    @Inject
+    internal lateinit var nightThemeRepository: NightThemeRepository
+
+    private val sessionsGooglePayViewModel: SessionsGooglePayViewModel by viewModels()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        // Helps to resize the view port when the keyboard is displayed.
+        WindowCompat.setDecorFitsSystemWindows(window, false)
+
+        // Insert return url in extras, so we can access it in the ViewModel through SavedStateHandle
+        val returnUrl = RedirectComponent.getReturnUrl(applicationContext) + "/sessions/googlepay"
+        intent = (intent ?: Intent()).putExtra(RETURN_URL_EXTRA, returnUrl)
+
+        setContent {
+            val useDarkTheme = when (nightThemeRepository.theme) {
+                NightTheme.DAY -> false
+                NightTheme.NIGHT -> true
+                NightTheme.SYSTEM -> isSystemInDarkTheme()
+            }
```

Comment:
What do you think of making an extension property or a function out of this?
This pattern will be used in every activity, which supports compose. E.g. `nightThemeRepository.shouldUseDarkTheme()` or `nightThemeRepository.useDarkTheme`